app/components/profile_picture_button.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import {Client4} from 'mattermost-redux/client';
+
+import {t} from 'app/utils/i18n';
+
+import AttachmentButton from 'app/components/attachment_button';
+
+export default class ProfilePictureButton extends PureComponent {
+    static propTypes = {
+        currentUser: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        extraOptions: PropTypes.func,
+    };
+
+    state = {
+        profileImageUri: null,
+    };
+
+    getProfileImageUri = async (id, lastPictureUpdate) => {
+        let uri;
+        try {
+            uri = await Client4.getProfilePictureUrl(id, lastPictureUpdate);
+        } catch (error) {
+            return null;
+        }
+        return this.setState({profileImageUri: uri});
+    }
+
+    isCustomProfileImage = async (id, lastPictureUpdate) => {
+        let uri;
+        let action = null;
+        try {
+            uri = await Client4.getProfilePictureUrl(id, lastPictureUpdate);
+        } catch (error) {
+            return null;
+        }
+        const {extraOptions} = this.props;
+        if (typeof extraOptions !== 'undefined') {
+            action = extraOptions;
+        }
+
+        if (uri.includes('?')) {
+            this.setState({
+                extraOptions:
+                    {
+                        action,
+                        text: {
+                            id: t('mobile.edit_profile.remove_profile_photo'),
+                            defaultMessage: 'Remove Photo',
+                        },
+                        textStyle: {
+                            color: '#CC3239',
+                        },
+                        icon: 'trash',
+                        iconStyle: {
+                            color: '#CC3239',
+                        },
+                    },
+            });
+        }
+        return this.setState({isCustomProfileImage: false});
```

Comment:
This can be removed since `this.state.isCustomProfileImage` isn't used. Also, I don't think `setState` has a return value, so I don't think we should be returning its result anyway.

Reply:
Ah, agreed. The docs imply there isn't anything returned like you said, and I'm not using that state anywhere.